### PR TITLE
Remove MIP parsing support and update test env

### PR DESCRIPTION
**Pull request by moonso**
This PR removes a module and a CLI command for parsing old mip files.
The test environment is also improved by more intuitive fixtures.

**Expected test outcome**:
- [ ] Behaviour should be the same

**Review:**
- [x] code approved by @Mropat 
- [x] tests executed by @moonso 
- [x] "Merge and deploy" approved by @moonso 
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
- [x] **MAJOR** - when you make incompatible API changes

### CODE REVIEW

File: tests/cli/store/test_cli_add_sex.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+def test_add_sex(cli_runner: CliRunner, sequence_db: Manager):
+    """Test to update sex for a sample with the CLI"""
+    # GIVEN an existing database with a sample without sex
+    sample_obj = Sample.query.first()
+    assert sample_obj.sex is None
+    sample_id = sample_obj.id
+
+    # WHEN adding information about the sex of the sample with the CLI
+    sex = "female"
+    result = cli_runner.invoke(add_sex, [sample_id, "-s", sex], obj={"db": sequence_db})
```
Comment (Mropat) [inline tests/cli/store/test_cli_add_sex.py:21]: Could context here be fixture instead?

File: tests/cli/store/test_cli_add_sex.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    sex = "male"
+    # WHEN updating the analysis sex determination
+    result = cli_runner.invoke(
+        add_sex, [sample_id, "-a", analysis_type, sex], obj={"db": sequence_db}
+    )
+
+    # THEN the CLI should be OK
+    assert result.exit_code == 0
+    # THEN assert it was communicated that the analysis did not exist
+    assert f"analysis not found: {sample_id}-{analysis_type}" in caplog.text
```
Comment (Mropat) [inline tests/cli/store/test_cli_add_sex.py:81]: This one is inconsistent with expected behavior from test_delete_nonexistent_analysis that expects exit code =! 1
Perhaps out of scope of this PR though!

**Review by Mropat — APPROVED** [review]
Cool tests as always! :sunglasses:  
Left some minor comments. 

Feel free to address some of those in another PR, so that this one can move forward!

File: tests/cli/store/test_cli_add_sex.py (hunk 2 above)
Comment (moonso) [reply tests/cli/store/test_cli_add_sex.py:81]: This is happening in a loop so I do not dare to change the behaviour atm. I guess one wants the loop to continue

File: tests/cli/store/test_cli_add_sex.py (hunk 1 above)
Comment (moonso) [reply tests/cli/store/test_cli_add_sex.py]: Fixed this

**Pull request merged**